src/main/webapp/app/shared/user-settings/ide-preferences/ide-settings.service.ts

```diff
@@ -10,8 +10,14 @@ export class IdeSettingsService {
     public ideSettingsUrl = 'api/ide-settings';
     error?: string;
 
-    constructor(private http: HttpClient) {
-        this.loadIdePreferences();
+    constructor(private http: HttpClient) {}
+
+    /**
+     * GET call to the server to receive the stored ide preferences of the current user
+     * @return the saved ide preference which were found in the database or error
+     */
+    public loadPredefinedIdes(): Observable<Ide[]> {
+        return this.http.get<Ide[]>(this.ideSettingsUrl + '/predefined');
```

Comment:
The naming of the method could be a bit more precise imo. Maybe `loadIdePreferences()` would fit better? It's not really predefined it has just been configured a while ago, right?

Reply:
yes exactly. this endpoints loads just some IdeDTO for displaying the options to choose from